komorebi.in

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import gi
+import signal
+import sys
+import logging
+
+gi.require_versions({
+	'Gtk': '3.0',
+	'GObject': '2.0',
+	'GtkClutter': '1.0',
+	'Clutter': '1.0',
+	'ClutterGst': '3.0',
+	'Gdk': '3.0',
+	'GdkPixbuf': '2.0',
+	'Gst': '1.0',
+	'WebKit2': '4.0',
+})
+
+from komorebi.preferences_window import PreferencesWindow, SETTINGS_CHANGED_SIGNAL
+import komorebi.settings
+import komorebi.utilities
+import komorebi.consts
+
+from komorebi.screen import Screen, SCREEN_SETTINGS_REQUEST_SIGNAL
+
+from gi.repository import Gio, Gtk, GtkClutter, Gdk, Gst, Clutter
+
+def check_desktop_compatible():
+	if 'wayland' in os.environ.get('XDG_SESSION_TYPE', '') or \
+		os.environ.get('WAYLAND_DISPLAY', ''):
+		return False
+
+	return True
+
+def main_parser():
+	parser = argparse.ArgumentParser()
+	parser.add_argument('-v', '--version',
+						action='store_true',
+						help='show current version')
+	parser.add_argument('-ss', '--single-screen',
+						action='store_true',
+						help='force komorebi to run only on the main screen')
+	parser.add_argument('-l', '--log',
+						type=str,
+						choices=['NORMAL', 'INFO', 'DEBUG'],
+						default='NORMAL',
+						help="set logging level for komorebi")
+
+	return parser
+
+def _on_destroy(*args):
+	logging.info("Quitting...")
+	Clutter.main_quit()
+
+def main():
+	print(f'Welcome to {komorebi.consts.package_name}')
+
+	# Handle Ctrl-C
+	signal.signal(signal.SIGINT, signal.SIG_DFL)
+
+	# Parse the arguments
+	parser = main_parser()
+	args = parser.parse_args()
+
+	if args.version:
+		print(f'Version: {komorebi.consts.package_version}\nMaintained by: Komorebi Team\n')
+		return
+
+	# Setup logger
+	log_level = logging.WARNING
+	log_format = ''
+
+	if args.log:
+		if args.log == 'INFO':
+			log_level = logging.INFO
+			log_format = '[%(levelname)s]: %(message)s'
+		elif args.log == 'DEBUG':
+			log_level = logging.DEBUG
+			log_format = '[%(levelname)s] (%(asctime)s): %(message)s'
+
+	logging.basicConfig(format=log_format, level=log_level, datefmt='%H:%M:%S')
+
+	# Ensure we are not on Wayland
+	if not check_desktop_compatible():
+		logging.error('Wayland detected. Not supported (yet) :(')
+		logging.info('Contribute to Komorebi and add the support! <3')
+		return
+
+	# Initialize backends
+	GtkClutter.init(sys.argv)
+	logging.debug("GtkClutter initialized")
+
+	komorebi.settings.load_settings()
+	logging.debug("Configuration file read")
+
+	logging.info('loading Gst')
+	Gst.init(sys.argv)
+	logging.debug('Gst initialized')
+
+	# Load resources
+	resource_path = os.path.join(komorebi.consts.package_datadir, 'komorebi.gresource')
+	resource = Gio.Resource.load(resource_path)
+	resource._register()
```

Comment:
Uh... This is private API, please don't use it. It may change anytime without warning, which will break our code. Using an underscore at the start of a name in Python indicates it is private API.

Reply:
Well, it's what's indicated to use in `Gio.Resource`'s [documentation](https://lazka.github.io/pgi-docs/#Gio-2.0/classes/Resource.html#Gio.Resource.load). But apparently [this](https://lazka.github.io/pgi-docs/Gio-2.0/functions.html#Gio.resources_register) works too, and looks less "internal" :slightly_smiling_face:

Reply:
Yeah, I'd say it is okay-ish then, but it could be a mistake in the documentation. Going with `resources_register` seems like the best option.

Reply:
Fixed